tests/build/build_above_src.srctree

```diff
@@ -0,0 +1,40 @@
+# Tests the out of place (ie with build_dir set) build when a src file is
+# "above" the setup.py (ie it's path begins with dots). 
+# For example, "../../../src/foo.pyx". See #2764 for details.
+
+cd a/b/c/d; PYTHON setup.py build_ext -b ../../pkg -t .
+PYTHON check_paths.py
+
+######## a/b/c/d/setup.py ########
+
+from distutils.core import setup
+from distutils.extension import Extension
+from Cython.Build import cythonize
+
+import numpy as np
+
+extensions = [
+    Extension(
+        'boolFunc',
+        ['../../../src/foo.pyx'],
+        include_dirs=[np.get_include()],
```

Comment:
Any reason you need to use numpy here? That makes the test fail (or requires it to be excluded) when numpy is not installed. Seems an unnecessary limitation to me.

Reply:
That's a mistake. It was left in from some code I based the test file on. I'll get rid of it.